CONTRIBUTING.rst suggestions (#28)
Ouranosinc/cookiecutter-pypackage: {{cookiecutter.project_slug}}/CONTRIBUTING.rst

    $ flit install --symlink
  {%- endif %}

  This installs ``{{ cookiecutter.project_slug }}`` in an "editable" state, meaning that changes to the code are immediately seen by the environment.

#. To ensure a consistent coding style, install the ``pre-commit`` hooks to your local clone::

    $ pre-commit install

>   On commit, ``pre-commit`` will check that{% if cookiecutter.use_black == 'y' %} ``black``, ``blackdoc``, ``isort``,{% endif %} ``flake8``, and ``ruff`` checks are passing, perform automatic fixes if possible, and warn of violations that require intervention. If your commit fails the checks initially, simply fix the errors and re-commit.

Reviewer: ```suggestion
  On commit, ``pre-commit`` will check that{% if cookiecutter.use_black == 'y' %} ``black``, ``blackdoc``, ``isort``,{% endif %} ``flake8``, and ``ruff`` checks are passing, perform automatic fixes if possible, and warn of violations that require intervention. If your commit fails the checks initially, simply fix the errors, re-add the files, and re-commit.
```

  Now you can make your changes locally.

#. When you're done making changes, we **strongly** suggest running the tests in your environment or with the help of ``tox``::

    $ python -m pytest
    # Or, to run multiple build tests
    $ tox

#. Commit your changes and push your branch to GitHub::

    $ git add .
    $ git commit -m "Your detailed description of your changes."
    $ git push origin name-of-your-bugfix-or-feature

  If ``pre-commit`` hooks fail, try re-committing your changes (or, if need be, you can skip them with `$ git commit --no-verify`).

#. Submit a `Pull Request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_ through the GitHub website.

#. When pushing your changes to your branch on GitHub, the documentation will automatically be tested to reflect the changes in your Pull Request. This build process can take several minutes at times. If you are actively making changes that affect the documentation and wish to save time, you can compile and test your changes beforehand locally with::

    # To generate the html and open it in your browser
    $ make docs

>     # To uniquely generate the html

Reviewer: Does `uniquely` work in that context? (says the Francophone to the Anglo 😅)
```suggestion
    # To only generate the html
```

Author: It's technically correct, but I think you're right that it's nearly a `francisisme` in English :laughing:

Reviewer: We're starting to corrupt you!

    $ make autodoc
    $ make -C docs html
    # To simply test that the docs pass build checks
    $ tox -e docs

#. Once your Pull Request has been accepted and merged to the ``main`` branch, several automated workflows will be triggered:

>     - The ``bump-version.yml`` workflow will automatically bump the patch version when pull requests are pushed to the ``main`` branch on GitHub. **It is not necessary to manually bump the version in your branch when merging (non-release) pull requests.**

Reviewer: "not necessary" or "not recommended"? If I manually bump within my PR, will `bump-my-version` be intelligent enough to understand it and not bump a second time?
```suggestion
    - The ``bump-version.yml`` workflow will automatically bump the patch version when pull requests are pushed to the ``main`` branch on GitHub. **It is thus not necessary to manually bump the version in your branch when merging (non-release) pull requests.**
```

Author: It will not be intelligent enough to know (and coding that would be _incredibly_ annoying to do).

Author: ```suggestion
    - The ``bump-version.yml`` workflow will automatically bump the patch version when pull requests are pushed to the ``main`` branch on GitHub. **It is not recommended to manually bump the version in your branch when merging (non-release) pull requests (this will cause the version to be bumped twice).**
```

    - `ReadTheDocs` will automatically build the documentation and publish it to the `latest` branch of `{{ cookiecutter.project_slug }}` documentation website.
    - If your branch is not a fork (ie: you are a maintainer), your branch will be automatically deleted.

  You will have contributed your first changes to ``{{ cookiecutter.project_slug }}``!

Pull Request Guidelines
-----------------------

Before you submit a pull request, check that it meets these guidelines:

> #. The pull request should include tests and should aim to provide `code coverage <https://en.wikipedia.org/wiki/Code_coverage>`_ all new lines of code. You can use the ``--cov-report html --cov {{ cookiecutter.project_slug }}`` flags during the call to ``pytest`` to generate an HTML report and analyse the current test coverage.

Reviewer: ```suggestion
#. The pull request should include tests and should aim to provide `code coverage <https://en.wikipedia.org/wiki/Code_coverage>`_ for all new lines of code. You can use the ``--cov-report html --cov {{ cookiecutter.project_slug }}`` flags during the call to ``pytest`` to generate an HTML report and analyse the current test coverage.
```
